examples/structured_2d_dgsem/elixir_shallowwater_well_balanced_wet_dry.jl

```diff
@@ -0,0 +1,235 @@
+
+indicator_sc = IndicatorHennemannGassner(equations, basis,
+                                         alpha_max=0.5,
+                                         alpha_min=0.001,
+                                         alpha_smooth=true,
+                                         variable=waterheight_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg=volume_flux,
+                                                 volume_flux_fv=surface_flux)
+
+solver = DGSEM(basis, surface_flux, volume_integral)
+
+
+###############################################################################
+# Create the StructuredMesh for the domain [0, 1]^2
+
+coordinates_min = (0.0, 0.0)
+coordinates_max = (1.0, 1.0)
+
+cells_per_dimension = (16, 16)
+
+mesh = StructuredMesh(cells_per_dimension, coordinates_min, coordinates_max)
+
+
+# create the semi discretization object
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+tspan = (0.0, 10.0)
+ode = semidiscretize(semi, tspan)
+
+###############################################################################
+# Workaround to set a discontinuous bottom topography for debugging and testing.
+
+# alternative version of the initial condition used to setup a truly discontinuous
+# bottom topography function for this academic testcase.
+# The errors from the analysis callback are not important but the error for this lake at rest test case
+# `∑|H0-(h+b)|` should be around machine roundoff
+# In contrast to the usual signature of initial conditions, this one get passed the
+# `element_id` explicitly. In particular, this initial conditions works as intended
+# only for the specific mesh created above!
+function initial_condition_discontinuous_well_balancedness(x, t, element_id, equations::ShallowWaterEquations2D)
+  # Set the background values
+  v1 = 0
+  v2 = 0
+  b = sin(4 * pi * x[1]) + 1
+  H = max(b, 1.5)
+
+  # very brute force approach to guarantee a discontinuous bottom on the StructuredMesh
+  if ((element_id >= 1   && element_id <= 8  ) ||
+      (element_id >= 17  && element_id <= 24 ) ||
+      (element_id >= 33  && element_id <= 40 ) ||
+      (element_id >= 49  && element_id <= 56 ) ||
+      (element_id >= 65  && element_id <= 72 ) ||
+      (element_id >= 81  && element_id <= 88 ) ||
+      (element_id >= 97  && element_id <= 104) ||
+      (element_id >= 113 && element_id <= 120) ||
+      (element_id >= 129 && element_id <= 136) ||
+      (element_id >= 145 && element_id <= 152) ||
+      (element_id >= 161 && element_id <= 168) ||
+      (element_id >= 177 && element_id <= 184) ||
+      (element_id >= 193 && element_id <= 200) ||
+      (element_id >= 209 && element_id <= 216) ||
+      (element_id >= 225 && element_id <= 232) ||
+      (element_id >= 241 && element_id <= 248))
```

Comment:
Would it make sense to pass some interior node instead of the element ID? Then, you could check the coordinates of the interior node, assuming that the elements are aligned with the discontinuity.

Reply:
I hadn't thought about that. I can try this to see if it cleans up this very brute force strategy of setting the discontinuity.